resources/persist/src/lib.rs

```diff
@@ -25,22 +33,77 @@ pub enum PersistError {
 #[derive(Serialize)]
 pub struct Persist;
 
-#[derive(Deserialize, Serialize, Clone)]
+#[derive(Debug, Deserialize, Serialize, Clone)]
 pub struct PersistInstance {
     service_name: ServiceName,
 }
 
 impl PersistInstance {
-    pub fn save<T: Serialize>(&self, key: &str, struc: T) -> Result<(), PersistError> {
-        let storage_folder = self.get_storage_folder();
-        fs::create_dir_all(storage_folder).map_err(PersistError::CreateFolder)?;
+    /// new method constructs a new PersistInstance along with its associated storage folder
+    pub fn new(service_name: ServiceName) -> Result<Self, shuttle_service::Error> {
+        let instance = Self { service_name };
+        let storage_folder = instance.get_storage_folder();
+        match fs::create_dir_all(storage_folder) {
+            Ok(_) => &instance,
+            Err(e) => {
+                return Err(shuttle_service::Error::Custom(
+                    PersistError::CreateFolder(e).into(),
+                ))
```

Comment:
My thought with converting to a `shuttle_service::Error::Custom` was to have that happen in `ResourceBuilder::output()` (since the trait requires a `shuttle_service::Error` type in that result). The `new` function can simply return this `PersistError` with `map_err` and `?`, similar to all other errors in this struct. The construction of `PersistInstance` in `output` should then use `::new(...)` and convert the error if it happens there (with this match clause or map_err).
(The code looks correct, it is just missing the usage of `new()` in the builder 😄)

Reply:
Got it.  Will sort it out this weekend.

Reply:
I think I got it!  Except for the failing cargo-shuttle circleci :(